src/common.rs

```diff
@@ -101,7 +101,31 @@ impl<'s> JsonPath<'s> {
     }
 }
 
-pub fn invoke<C: FromIterator<Option<I>> + 'static, I>(
+/// Very similar to `FromIterator` but we define it here so we can implement it as we wish,
```

Comment:
@davidhewitt might have a better idea of how to do this neatly, but this works.

The point is that `FromIterator<Option<I>` which we previously used isn't implemented for `ListArray` with `Vec<String>`.

Confusingly I tried implementing `FromOptionIter` for everything that implements `FromIterator`, but then the implementation for `ListArray` claimed to conflict with an existing implementation 🤷.